DiabloII.Application/Controllers/SuggestionsController.cs

```diff
@@ -34,8 +39,20 @@ public SuggestionsController(ISuggestionReader reader, ISuggestionCommandHandler
         [Route("suggestions")]
         [HttpGet]
         [ProducesResponseType(typeof(IReadOnlyCollection<SuggestionDto>), StatusCodes.Status200OK)]
-        public ActionResult<IReadOnlyCollection<SuggestionDto>> GetAll() =>
-            GetAll(_reader, _mapper);
+        public ActionResult<IReadOnlyCollection<SuggestionDto>> GetAll()
+        {
+            var response = _reader
+                .GetAll()
+                .Select(dataModel =>
+                {
+                    var dto = _mapper.Map<SuggestionDto>(dataModel);
+
+                    return _suggestionHalService.AddLinks(dto);
+                })
+                .ToList();
+
+            return Ok(response);
```

Comment:
This logic of reader.GetAll that returns HalReponse should be extracted in the BaseController, this makes the application messy, I don't like that.